Client/src/Unified/Services/Submitter/BatchUntilInactiveBlock.cs

```diff
@@ -0,0 +1,249 @@
+// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Dataflow;
+
+using JetBrains.Annotations;
+
+namespace ArmoniK.DevelopmentKit.Client.Unified.Services.Submitter;
+
+/// <summary>
+///   Provides a dataFlow block that batches inputs into arrays.
+///   A batch is produced when the number of currently queued items becomes equal
+///   to BatchSize, or when a Timeout period has elapsed after receiving the last item.
+/// </summary>
+public class BatchUntilInactiveBlock<T> : IPropagatorBlock<T, T[]>, IReceivableSourceBlock<T[]>
+{
+  private readonly ExecutionDataflowBlockOptions executionDataFlowBlockOptions_;
+  private readonly BatchBlock<T>                 source_;
+  private readonly TransformBlock<T[], T[]>      timeoutTransformBlock_;
+  private readonly Timer                         timer_;
+
+  /// <summary>
+  ///   The buffer construct base on the number of request in the buffer
+  ///   Be aware that buffer should be T sized for network B/W
+  /// </summary>
+  /// <param name="bufferRequestsSize"></param>
+  /// <param name="timeout">Time out before the next submit call</param>
+  /// <param name="dataFlowBlockOptions">
+  ///   Options to configure message.
+  ///   (https://learn.microsoft.com/fr-fr/dotnet/api/system.threading.tasks.dataflow.groupingdataflowblockoptions?view=net-6.0)
+  /// </param>
+  /// <param name="executionDataFlowBlockOptions">Parameters to control execution for each block in pipeline</param>
+  public BatchUntilInactiveBlock(int                                       bufferRequestsSize,
+                                 TimeSpan                                  timeout,
+                                 [CanBeNull] ExecutionDataflowBlockOptions executionDataFlowBlockOptions = null)
```

Comment:
Can't you use `?` instead of `[CanBeNull]`?

Reply:
Nullable is not enabled

Reply:
Specific issue here https://github.com/aneoconsulting/ArmoniK/issues/647